### ISSUE TITLE
waterfox can't build under musl-libc, ignore --target patameter

### ISSUE THREAD

**Issue by ghost** [BrowserWorks/waterfox#513]
a try to build it under an musl-libc with the config parameter:

```
echo "ac_add_options --host=x86_64-unknown-linux-musl" >>.mozconfig
echo "ac_add_options --target=x86_64-unknown-linux-musl" >>.mozconfig
``` 
as like for firefox, brings:

```
 0:02.32 checking MOZ_FFI_LIBS... -L/usr/lib/../lib -lffi
 0:02.32 checking for rustc... /usr/bin/rustc
 0:02.32 checking for cargo... /usr/bin/cargo
 0:02.36 checking rustc version... 1.25.0
 0:02.38 checking cargo version... 0.26.0
 0:02.46 DEBUG: Executing: `/usr/bin/rustc --crate-type staticlib --target=x86_64-unknown-linux-gnu -o /tmp/conftestKInzJ6.rlib /tmp/conftestIaKqWs.rs`
 0:02.46 DEBUG: The command returned non-zero exit status 101.
 0:02.46 DEBUG: Its error output was:
 0:02.46 DEBUG: | error[E0463]: can't find crate for `std`
 0:02.46 DEBUG: |   |
 0:02.46 DEBUG: |   = note: the `x86_64-unknown-linux-gnu` target may not be installed
 0:02.46 DEBUG: |
 0:02.46 DEBUG: | error: aborting due to previous error
 0:02.46 DEBUG: |
 0:02.46 ERROR: Cannot compile for x86_64-unknown-linux-musl with /usr/bin/rustc
 0:02.46 The target may be unsupported, or you may not have
 0:02.46 a rust std library for that target installed. Try:
```

**Comment by jbeich** [comment]
Does backporting mozilla/gecko-dev@6a8726dd5f98 help?

**Comment by ghost** [comment]
with this patch, the build proccess starts

**Comment by ghost** [comment]
after a while it stops with:

```
18:38.13 /builddir/Waterfox-56.1.0/obj-x86_64-unknown-linux-musl/dist/include/mozilla/mozalloc.h:180:52: warning: dynamic exception specifications are deprecated in C++11 [-Wdeprecated]
18:38.13  #define MOZALLOC_THROW_BAD_ALLOC_IF_HAS_EXCEPTIONS throw(std::bad_alloc)
18:38.13                                                     ^
18:38.13 /builddir/Waterfox-56.1.0/obj-x86_64-unknown-linux-musl/dist/include/mozilla/mozalloc.h:183:34: note: in expansion of macro 'MOZALLOC_THROW_BAD_ALLOC_IF_HAS_EXCEPTIONS'
18:38.13  #define MOZALLOC_THROW_BAD_ALLOC MOZALLOC_THROW_BAD_ALLOC_IF_HAS_EXCEPTIONS
18:38.13                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
18:38.13 /builddir/Waterfox-56.1.0/obj-x86_64-unknown-linux-musl/dist/include/mozilla/mozalloc.h:204:35: note: in expansion of macro 'MOZALLOC_THROW_BAD_ALLOC'
18:38.13  void* operator new[](size_t size) MOZALLOC_THROW_BAD_ALLOC
18:38.13                                    ^~~~~~~~~~~~~~~~~~~~~~~~
18:41.49 libaccessible_atk.a.desc
18:41.54 make[3]: *** [/builddir/Waterfox-56.1.0/config/recurse.mk:33: compile] Error 2
18:41.54 make[2]: *** [/builddir/Waterfox-56.1.0/config/rules.mk:453: default] Error 2
18:41.54 make[1]: *** [/builddir/Waterfox-56.1.0/client.mk:419: realbuild] Error 2
18:41.54 make: *** [client.mk:170: build] Error 2
18:41.54 0 compiler warnings present.
Error running mach:

    ['build']

The error occurred in code that was called by the mach command. This is either
a bug in the called code itself or in the way that mach is calling it.

You should consider filing a bug for this issue.

If filing a bug, please include the full output of mach, including this error
message.

The details of the failure are as follows:

AssertionError

  File "/builddir/Waterfox-56.1.0/python/mozbuild/mozbuild/mach_commands.py", line 595, in build
    monitor.finish(record_usage=status==0)
  File "/builddir/Waterfox-56.1.0/python/mozbuild/mozbuild/mach_commands.py", line 284, in __exit__
    self.monitor.stop_resource_recording()
  File "/builddir/Waterfox-56.1.0/python/mozbuild/mozbuild/controller/building.py", line 258, in stop_resource_recording
    self.resources.stop()
  File "/builddir/Waterfox-56.1.0/testing/mozbase/mozsystemmonitor/mozsystemmonitor/resourcemonitor.py", line 289, in stop
    assert self._running
```

**Comment by jbeich** [comment]
Sorry, I can't help unless you quote the actual error or provide full build log.

**Comment by MrAlex94** [comment]
Yes, please post the full log.

**Comment by ghost** [comment]
afer restart the build procces the log is:

https://paste.pound-python.org/show/uyDYTK0FQ33Z00rviGLP/

**Comment by jbeich** [comment]
https://paste.pound-python.org/show/uyDYTK0FQ33Z00rviGLP/ has:
> ```rust
>  0:10.54 error: doc comment not used by rustdoc
>  0:10.55    --> /builddir/Waterfox-56.1.0/servo/components/style/values/specified/image.rs:689:17
>  0:10.55     |
>  0:10.55 689 |                 /// `to` keyword is mandatory in modern syntax.
>  0:10.55     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  0:10.55     |
>  0:10.55 note: lint level defined here
>  0:10.55    --> /builddir/Waterfox-56.1.0/servo/components/style/lib.rs:26:9
>  0:10.55     |
>  0:10.55 26  | #![deny(warnings)]
>  0:10.55     |         ^^^^^^^^
>  0:10.55     = note: #[deny(unused_doc_comment)] implied by #[deny(warnings)]
>  0:10.55 
>  0:10.55 error: doc comment not used by rustdoc
>  0:10.55    --> /builddir/Waterfox-56.1.0/servo/components/style/values/specified/image.rs:695:17
>  0:10.55     |
>  0:10.55 695 |                 /// There is no `to` keyword in webkit prefixed syntax. If it's consumed,
>  0:10.55     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  0:10.55 
>  0:10.69 error: doc comment not used by rustdoc
>  0:10.69     --> /builddir/Waterfox-56.1.0/obj-x86_64-unknown-linux-musl/toolkit/library/x86_64-unknown-linux-musl/release/build/style-44dce245ab397a04/out/gecko_properties.rs:9131:9
>  0:10.69      |
>  0:10.69 9131 |         /// XXXManishearth this should also handle text zoom
>  0:10.69      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>  0:10.69 
> [...]
>  0:13.23 error: aborting due to 5 previous errors
>  0:13.23 
>  0:13.39 error: Could not compile `style`.
>  0:13.39 
>  0:13.39 To learn more, run the command again with --verbose.
>  0:13.39 make[5]: *** [/builddir/Waterfox-56.1.0/config/rules.mk:987: force-cargo-library-build] Error 101
>  0:13.39 make[4]: *** [/builddir/Waterfox-56.1.0/config/recurse.mk:73: toolkit/library/rust/target] Error 2
>  0:13.39 make[4]: *** Waiting for unfinished jobs....
> ```

Same as #477. To fix apply #508.

**Comment by ghost** [comment]
hmm, to match patches for a release version, waiting for the next release.
